### ISSUE TITLE
Consider using pickle protocol 5 instead of joblib when possible

### ISSUE THREAD

**Issue by victornoel** [jpmml/sklearn2pmml#290]
Hi,

In my exploration with #287, I understood that using joblib to dump a pickle could be replaced by using pickle with protocol 5 directly (see the note in https://joblib.readthedocs.io/en/latest/persistence.html).

I'm not sure what impact it will exactly have, but it seems to be more elegant and most certainly will be more efficient than using joblib for big objects.

**Comment by vruusmann** [comment]
The Scikit-Learn team is recommending joblib (over pickle) for model persistence:
https://scikit-learn.org/stable/modules/model_persistence.html

> I'm not sure what impact it will exactly have, but it seems to be more elegant and most certainly will be more efficient than using joblib for big objects.

Joblib works better with data that is typical with Scikit-Learn models (many smallish Numpy arrays).

The fact that you're desperately trying to force a 2 GB JSON file into a pickle file is not a representative use case.

**Comment by victornoel** [comment]
@vruusmann I wasn't asking in the context of my crazy example, just for more general use cases.

I think the point of the note in the joblib page is that since protocol 5, it's not needed to rely on joblib when using numpy arrays, hence my proposal :)

The sklean page actually talk about large numpy arrays:
> In the specific case of scikit-learn, it may be better to use joblib’s replacement of pickle (dump & load), which is more efficient on objects that carry large numpy arrays internally as is often the case for fitted scikit-learn estimators

**Comment by victornoel** [comment]
That's also what https://stackoverflow.com/a/12617603 is saying apparently (and also that pickle is better for small numpy arrays).

One advantage of joblib is still memory mapping, not sure if that can be used in sklearn2pmml context though.
